controller/tests/array_action/ds8k/test_array_mediator_ds8k.py

```diff
@@ -192,17 +194,90 @@ def test_delete_volume_with_flashcopies_as_source_and_target_failed(self):
                    }),
             Munch({"sourcevolume": "0003",
                    "targetvolume": "0001",
                    "id": "0003:0001",
-                   "state": "valid"
+                   "state": "valid",
+                   "backgroundcopy": "enabled",
+                   "representation": {}
                    })]
+        self.client_mock.get_flashcopies.return_value = Munch({"out_of_sync_tracks": "0"})
         self.client_mock.delete_volume.side_effect = ClientException("500")
-        with self.assertRaises(array_errors.VolumeDeletionError):
+        with self.assertRaises(array_errors.ObjectIsStillInUseError):
             self.array.delete_volume("0001")
-        self.client_mock.delete_flashcopy.assert_called_once_with("0003:0001")
+
+    def test_delete_volume_with_source_no_flashcopies_deleted(self):
+        self.client_mock.get_volume.return_value = Munch(
+            {"cap": "1073741824",
+             "id": "0001",
+             "name": "test_name",
+             "pool": "fake_pool",
+             "tp": "ese",
+             "flashcopy": ""
+             }
+        )
+        self.client_mock.get_flashcopies_by_volume.return_value = [
+            Munch({"sourcevolume": "0001",
+                   "targetvolume": "0002",
+                   "id": "0001:0002",
+                   "state": "valid",
+                   "backgroundcopy": "disabled",
+                   "representation": {}
+                   })]
+        self.client_mock.get_flashcopies.return_value = Munch({"out_of_sync_tracks": "0"})
+        self.client_mock.delete_volume.side_effect = ClientException("500")
+        with self.assertRaises(array_errors.ObjectIsStillInUseError):
+            self.array.delete_volume("0001")
+
+    def test_delete_volume_with_flashcopy_still_copying(self):
+        self.client_mock.get_volume.return_value = Munch(
+            {"cap": "1073741824",
+             "id": "0001",
+             "name": "test_name",
+             "pool": "fake_pool",
+             "tp": "ese",
+             "flashcopy": ""
+             }
+        )
+        self.client_mock.get_flashcopies_by_volume.return_value = [
+            Munch({"sourcevolume": "0001",
+                   "targetvolume": "0002",
+                   "id": "0001:0002",
+                   "state": "valid",
+                   "backgroundcopy": "enabled"
+                   })]
+        self.client_mock.get_flashcopies.return_value = Munch({"out_of_sync_tracks": "55",
+                                                               "targetvolume": "0002",
+                                                               "representation": {}})
+        self.client_mock.delete_volume.side_effect = ClientException("500")
+        with self.assertRaises(array_errors.ObjectIsStillInUseError):
+            self.array.delete_volume("0001")
+        self.client_mock.delete_flashcopy.assert_not_called()
+
+    def test_delete_volume_with_source_flashcopies_deleted(self):
```

Comment:
go over the controller server volume clone PR and make sure there are no [common issues](https://github.com/IBM/ibm-block-csi-driver/pull/233#discussion_r497731184)

Reply:
done

Reply:
are you sure? I don't see any change here

Reply:
oh, look like I missed that one, I guess i used it as an anchor and forgot about it..

Reply:
done